### PR TITLE
2d data matrix barcode

### PR DESCRIPTION
**Pull request by betzrhodes**

### CODE REVIEW

**Comment by ReshefElisha** [comment]
Made small change to remove redundant command. Otherwise looks good to me.

**Comment by betzrhodes** [comment]
This has been sitting for a while. We have been using internally to print labels for quite some time so it is definitely ready for a release. I've updated the master for an initial release, and cleaned up this pull request to incorporate support for data matrix bar codes.  Please review so we can release either v0.1.0 (the version sitting on master) and/or v0.2.0 the one in this pull request.

**Review by ppetrosh — APPROVED** [review]

File: QL720NW.device.lib.nut [hunk 1]
```diff
@@ -314,6 +358,38 @@ class QL720NW {
...
+        if (!("code_number" in config)) { config.code_number <- 0; }
+        if (!("num_partitions" in config)) { config.num_partitions <- 0; }
+        if (!("parity_data" in config)) { config["parity_data"] <- 0; }
+        if (!("error_correction" in config)) { config["error_correction"] <- QL720NW_BARCODE_2D_QR_ERROR_CORRECTION_STANDARD; }
+        if (!("data_input_method" in config)) { config["data_input_method"] <- QL720NW_BARCODE_2D_QR_DATA_INPUT_AUTO; }
+
+        // Check ranges
+        if (config.structured_append_partitioned) {
+            config.structured_append <- QL720NW_BARCODE_2D_QR_STRUCTURE_PARTITIONED;
+            if (config.code_number < 1 || config.code_number > 16) throw "Unknown code number";
```
Comment (ppetrosh) [inline QL720NW.device.lib.nut:382]: Unknown->Invalid

File: QL720NW.device.lib.nut [hunk 2]
```diff
@@ -314,6 +358,38 @@ class QL720NW {
...
+            if (config.num_partitions < 2 || config.num_partitions > 16) throw "Unknown number of partitions";
+        } else {
+            config.structured_append <- QL720NW_BARCODE_2D_QR_STRUCTURE_NOT_PARTITIONED;
+            config.code_number = "\x00";
```
Comment (ppetrosh) [inline QL720NW.device.lib.nut:386]: Why these hex values here and not at the beginning of the method?

**Pull request merged**